tasks/test_results_processor.py

```diff
@@ -179,156 +180,219 @@ def _bulk_write_tests_to_db(
-                        "avg_duration_seconds": duration_seconds,
-                        "pass_count": 1 if outcome == str(Outcome.Pass) else 0,
-                        "fail_count": 1
-                        if outcome == str(Outcome.Failure)
-                        or outcome == str(Outcome.Error)
-                        else 0,
-                        "skip_count": 1 if outcome == str(Outcome.Skip) else 0,
-                        "flaky_fail_count": 1
-                        if test_id in flaky_test_set
-                        and (
-                            outcome == str(Outcome.Failure)
-                            or outcome == str(Outcome.Error)
-                        )
-                        else 0,
-                        "branch": branch,
-                        "date": date.today(),
-                        "latest_run": datetime.now(),
-                        "commits_where_fail": [commitid]
-                        if (
-                            outcome == str(Outcome.Failure)
-                            or outcome == str(Outcome.Error)
-                        )
-                        else [],
-                    }
-
                 if outcome != str(Outcome.Skip):
                     if test_id in daily_totals:
-                        update_daily_total()
+                        self.update_daily_totals(
+                            daily_totals, test_id, duration_seconds, outcome
+                        )
                     else:
-                        create_daily_total()
+                        self.create_daily_totals(
+                            daily_totals,
+                            test_id,
+                            repoid,
+                            duration_seconds,
+                            outcome,
+                            branch,
+                            commitid,
+                            flaky_test_set,
+                        )
 
         # Upsert Tests
         if len(test_data) > 0:
-            test_insert = insert(Test.__table__).values(
-                sorted(
-                    test_data.values(),
-                    key=lambda x: str(x["id"]),
-                )
+            sorted_tests = sorted(
+                test_data.values(),
+                key=lambda x: str(x["id"]),
             )
-            insert_on_conflict_do_update = test_insert.on_conflict_do_update(
-                index_elements=["repoid", "name", "testsuite", "flags_hash"],
-                set_={
-                    "framework": test_insert.excluded.framework,
-                    "computed_name": test_insert.excluded.computed_name,
-                    "filename": test_insert.excluded.filename,
-                },
-            )
-            db_session.execute(insert_on_conflict_do_update)
-            db_session.commit()
+            self.save_tests(db_session, sorted_tests)
 
         log.info("Upserted tests to database", extra=dict(upload_id=upload_id))
```

Comment:
all these logs here and below should probably be nested in the  `if` they belong to.